hebiros/include/hebiros/hebiros.cpp

```diff
@@ -208,8 +208,290 @@ bool Hebiros_Node::srv_add_group_from_urdf(
+
+  for (int i = 0; i < data.name.size(); i++) {
+    if (group_joints[group_name].find(data.name[i]) != group_joints[group_name].end()) {
+
+      int joint_index = group_joints[group_name][data.name[i]];
+
+      if (i < data.kp.size()) {
+        (*group_command)[joint_index].
+          settings().actuator().positionGains().positionKp().set(data.kp[i]);
+      }
+      if (i < data.ki.size()) {
+        (*group_command)[joint_index].
+          settings().actuator().positionGains().positionKi().set(data.ki[i]);
+      }
+      if (i < data.kd.size()) {
+        (*group_command)[joint_index].
+          settings().actuator().positionGains().positionKd().set(data.kd[i]);
+      }
+      if (i < data.feed_forward.size()) {
+        (*group_command)[joint_index].
+          settings().actuator().positionGains().positionFeedForward().set(data.feed_forward[i]);
+      }
+      if (i < data.dead_zone.size()) {
+        (*group_command)[joint_index].
+          settings().actuator().positionGains().positionDeadZone().set(data.dead_zone[i]);
+      }
+      if (i < data.i_clamp.size()) {
+        (*group_command)[joint_index].
+          settings().actuator().positionGains().positionIClamp().set(data.i_clamp[i]);
+      }
+      if (i < data.punch.size()) {
+        (*group_command)[joint_index].
+          settings().actuator().positionGains().positionPunch().set(data.punch[i]);
+      }
+      if (i < data.min_target.size()) {
+        (*group_command)[joint_index].
+          settings().actuator().positionGains().positionMinTarget().set(data.min_target[i]);
+      }
+      if (i < data.max_target.size()) {
+        (*group_command)[joint_index].
+          settings().actuator().positionGains().positionMaxTarget().set(data.max_target[i]);
+      }
+      if (i < data.target_lowpass.size()) {
+        (*group_command)[joint_index].
+          settings().actuator().positionGains().positionTargetLowpass().set(data.target_lowpass[i]);
+      }
+      if (i < data.min_output.size()) {
+        (*group_command)[joint_index].
+          settings().actuator().positionGains().positionMinOutput().set(data.min_output[i]);
+      }
+      if (i < data.max_output.size()) {
+        (*group_command)[joint_index].
+          settings().actuator().positionGains().positionMaxOutput().set(data.max_output[i]);
+      }
+      if (i < data.output_lowpass.size()) {
+        (*group_command)[joint_index].
+          settings().actuator().positionGains().positionOutputLowpass().set(data.output_lowpass[i]);
+      }
+      if (i < data.d_on_error.size()) {
+        (*group_command)[joint_index].
+          settings().actuator().positionGains().positionDOnError().set(data.d_on_error[i]);
+      }
+    }
+    else {
+      ROS_WARN("Unable to find joint: %s.  Command will not be sent.", data.name[i].c_str());
+    }
+  }
+}
+
+//Add velocity gain values to a group command
+void Hebiros_Node::add_velocity_gains_command(GroupCommand* group_command,
```

Comment:
With the new C++ API that is coming, we'll be able to easily condense position/velocity/effort code into a single function -- sorry this is so monotonous right now!